Submit border patrol data to Tidy Tuesday

gdatascience opened the pull request:
Thank you for your submission! Please leave this text as-is when you submit, and then check the checklist.

Usability:

- [x] This dataset has not already been used in TidyTuesday.
- [x] The dataset is less than 20MB when saved as a tidy CSV.
- [x] I can imagine at least one data visualization related to this dataset.

Preparation:

- [x] `cleaning.R` has code to download and clean the dataset, resulting in descriptively-named data.frames.
- [x] I ran `saving.R` to create `csv`s and `md` data dictionaries for each of my data.frames.
- [x] I filled in descriptions for each variable in each of the `{dataset}.md` files.
- [x] I edited the `intro.md` file to introduce my dataset.
- [x] I included at least one image for my dataset as a `png` file.
- [x] I completed the information in `meta.yaml`, including descriptive alt text for each image.
- [x] I provided information in `meta.yaml` about how to credit me, and deleted any parts of that block that I do not want you to use.

jonthegeek commented on data/curated/border-patrol/cbp_resp.md:
```diff
@@ -0,0 +1,14 @@
+|aor_abbv               |character |The field office or sector where the encounter occurred (abbreviated) |
+|demographic            |character |Categories under which individuals were encountered based on factors such as age, admissibility, and relationship |
+|citizenship            |character |Citizenship of the individual encountered |
+|title_of_authority     |character |The authority under which the noncitizen was processed |
```
It looks like you partially explain in the next bullet, but let's make sure it's clear!

jonthegeek commented on data/curated/border-patrol/intro.md:
```diff
@@ -0,0 +1,13 @@
+DESCRIPTION
+
```
```suggestion
```
This is a placeholder for the place to add your description. I think I'll make a video for each of these sections one of these days, because it's difficult to express some of this without showing an example!

jonthegeek commented on data/curated/border-patrol/meta.yaml:
```diff
@@ -0,0 +1,19 @@
+title: U.S. Customs and Border Protection (CBP) encounter data
```
```suggestion
title: U.S. Customs and Border Protection (CBP) Encounter Data
```
I like to use Title Case for these.

jonthegeek commented on data/curated/border-patrol/intro.md:
```diff
@@ -0,0 +1,13 @@
+
+> Encounter data includes U.S. Border Patrol Title 8 apprehensions, Office of Field Operations Title 8 inadmissibles, and all Title 42 expulsions for fiscal years 2020 to date. Data is available for the Northern Land Border, Southwest Land Border, and Nationwide (i.e., air, land, and sea modes of transportation) encounters. Data is extracted from live CBP systems and data sources. Statistical inf ...
+
+Thank you to [Tony Galván](https://www.linkedin.com/in/anthony-raul-galvan/) for curating this dataset and providing a [blog post](https://posit.cloud/content/9130042) that explores the data in more detail.
```
For both here and meta.yaml: I can't view the content at this URL. I think this is your edit URL, not the published URL. I can't remember how to make content from posit.cloud public, but can help you sort it out if you need someone else to poke around! I recommend loading the URL in an incognito browser window (ctrl+shift+N) to see if someone who isn't logged in as you has access. I created #771 to make sure I check this every time, because I *almost* missed this!

gdatascience-acorns replied:
I changed access from private to "All Posit Cloud Users". I'm not sure there is a way to make it truly open to the "public"

jonthegeek replied:
Doh! Still not available, even when logged in: 
> Content Unavailable
> Publishing on Posit Cloud has been deprecated and previously published data documents, applications, and APIs are no longer accessible.
> 
> RStudio and Jupyter code projects are unaffected by this change.
> 
> For more information, please see [our message about this change](https://docs.posit.co/cloud/whats_new/#sunsetting-posit-cloud-publishing).

Maybe link to one of the LinkedIn posts? It'd be awesome if you could migrate it to a public blog, but I understand that that can be a pretty big lift!

jonthegeek replied:
(see above, but I'm keeping this comment unresolved so we remember to update both places)

jonthegeek commented on data/curated/border-patrol/meta.yaml:
```diff
@@ -0,0 +1,19 @@
+# Please include at least one image, and up to three images
+- file: inadmissible-aliens-plot.png
+  alt: >
+    A bar plot of monthly inadmissible CHNV (citizens of Cuba, Haiti, Nicaragua, and Venezuela) aliens encountered by U.S. Border Patrol agents from 2020 to 2024.
```
Add a source (you?). Also describe the plot content. What does it show? This guide doesn't 100% apply to plots, but it's a good place to start: https://bsky.app/profile/ind3fatigable.xyz/post/3lb5weqbnt22t
A tip I heard that stuck with me is that alt text should serve as a *replacement* for the image, not only a *description* of the image.

jonthegeek replied:
Did you create the plot? I see you added a credit for the data, but I meant a credit for the image. Note: Looking at past examples won't necessarily show a great track record about this, but it's something I want us to get better about!

jonthegeek commented on data/curated/border-patrol/meta.yaml:
```diff
@@ -0,0 +1,19 @@
+credit:
+# We want to thank you for curating this dataset! If you do not want a 
+# particular type of credit, please delete the related line.
+  post: Tony Galván
```
I *think* I use parts of this to auto-credit you once I use a command to "assign" this to a week, so we might have an edit to do in the post to make sure we don't double up. No worries, but let's remember to check that!

I don't have it in the template yet, but are you on bluesky yet? If so, add a `bluesky:` line. If not... [join bluesky](https://bsky.app/)! It's where all the fun #RStats conversations are happening! And then add a `bluseky:` line 😁

jonthegeek reviewed (changes requested):
Thank you so much! This is really close, just a few tweaks here and there. It's so nice to be able to take the time to perfect this, rather than slapping something together at the last minute like I've had to do a few times recently 🙃

Definitely let me know if there's anything that was particularly confusing in the instructions, particularly things that you thought you might have done wrong but it turned out you were right. I want to instill as little fear of failure/impostor syndrome as I can!

I don't know your comfort level/experience with PRs, so in case you aren't sure what's next:
- Click "Commit suggestion" on anything that you agree with in my review, to apply that change to your submission.
- On your machine, Pull the changes from github to your local copy. You can click "Pull" in the RStudio "Git" tab, run `usethis::pr_pull()`, or use the command line to `git pull` the updates. Or technically you can probably use the web UI for the things that are left to change, but I'm always more comfortable working locally.
- Make the additional edits described. Let me know if you need help with anything!
- "Push" the changes up to github ("Git" tab, `usethis::pr_push()`, or `git push`).
- (If you can; you might not have permission to do this) Click the circle-arrow next to my name under "Reviewers". That isn't really necessary here (I get alerts about every change in this repo), but it's good practice for PRs in general, to let the reviewer know "Ok, I did what you asked!"

Again, thank you SO MUCH for this! It's super helpful!

gdatascience-acorns commented:
@jonthegeek Thanks for the review and suggestions. I do not have much experience with GitHub (even though I have 2 accounts: 1 personal and 1 for work). My personal account is what I use for my Tidy Tuesday work, but I also connected my work account to RStudio to use CoPilot. I think this is how I got messed up. I don't think my stuff ended up on the fork I created. Maybe that's why I can't see "Commit suggestion". Here's what I see for the first suggestion:
![image](https://github.com/user-attachments/assets/2ccdf12f-65bf-410a-b6c6-0dd8441c9d96)
Any advice on how to proceed?

gdatascience commented:
Doh! I need to make my profile pictures different. I was logged in to GitHub with my work account (which is why I couldn't see the "Commit suggestion" button. I made the changes manually and committed them. Hopefully that works.

jonthegeek reviewed (changes requested):
The post still isn't accessible, unfortunately. Let's figure out something people can see! Make sure you aren't logged in as acorns! 😁

gdatascience commented:
> The post still isn't accessible, unfortunately. Let's figure out something people can see! Make sure you aren't logged in as acorns! 😁

I moved the article to GitHub pages and updated the alt text. Now that I have GitHub pages set up, I might have to start working on cleaning the website.

jonthegeek reviewed (commented):
Looks great! Now I need to assign it to a week, and make sure nothing strange happens in that process...

jonthegeek commented:
/assign border-patrol 2024-11-26

jonthegeek commented on data/2024/2024-11-26/readme.md:
```diff
@@ -0,0 +1,90 @@
+-   How has the implementation (and potential end) of Title 42 affected migration and enforcement trends compared to Title 8 actions?
+-   What are the key differences in migration patterns and enforcement activity between the Northern and Southwest Land Borders?
+-   Are there seasonal or year-over-year trends in encounters that can help predict future migration patterns?
+
+Thank you to [Tony Galván](https://github.com/gdatascience) for curating this week's dataset.
```
```suggestion
```
Delete duplicate credit.

jonthegeek commented on data/2024/2024-11-26/readme.md:
````diff
@@ -0,0 +1,90 @@
+### Cleaning Script
+
+```r
+cbp_resp <- bind_rows(
````
```suggestion
library(tidyverse)
library(janitor)

cbp_resp <- bind_rows(
```
Oops, I like to make sure we show packages used at the start.

jonthegeek reviewed (approved):
Just a couple small tweaks, which I'll accept and merge. Thanks!
